low-level/files/opt/seagate/sspl/setup/resource_map/lr2/storage.py

```diff
@@ -68,40 +74,99 @@ def get_health_info(self, rpath):
         """
         Fetch health information for given FRU.
 
-        rpath: Resouce path (Example: nodes[0]>storage[0]>hw>controllers)
+        rpath: Resource path (Example: node>storage[0]>hw>controllers)
         """
         info = {}
-        fru_found = False
+        resource_found = False
         nodes = rpath.strip().split(">")
         leaf_node, _ = self.get_node_details(nodes[-1])
+
+        # Fetch health information for all sub nodes
         if leaf_node == "storage":
-            for fru in self.storage_frus:
-                try:
-                    info.update({fru: self.storage_frus[fru]()})
-                except:
-                    # TODO: Log the exception
-                    info.update({fru: None})
-            info["last_updated"] = int(time.time())
-            fru_found = True
+            info = self.get_storage_health_info()
+            resource_found = True
```

Comment:
At any stage, if function before below statement fails, then this resource_found wont get true. Feel should keep it under try-exception block , line 85-106
>resource_found = True

Reply:
You are right.
I'm thinking to put this immediate to the check rather than keeping it after fetching health info.

Reply:
Done.